lib/IR/GraphScheduler.cpp

```diff
@@ -140,6 +140,26 @@ class ChildMemSizeBasedScheduler : public Scheduler {
       orderedChildren.push_back(N->getPredicate());
     }
 
+    // SaveNode hack:
+    // We don't model memory dependencies, but we still need to honor them.
+    // Make sure the SaveNode happens after the last use of the output variable.
+    if (SaveNode *save = llvm::dyn_cast<SaveNode>(N)) {
```

Comment:
Use `auto *save` as we usually use this style with `if (.. dyn_cast<Ty>(expr))` in our code?